### ISSUE TITLE
ipopt on Mac for MPC project

### ISSUE THREAD

**Issue by thelukasssheee** [udacity/sdc-issue-reports#1412]
Hey guys,

very frustrating issues while trying to install ipopt... tried all the hints and issues that were available on Slack and here in the issue report. 

Install of ipopt is failing, because it always finds at some mount multiple formulae. What I do not get: even when I untap the doubled taps before install, brew seems to resurrect them, as shown below.  

I'm working on MacOS 10.13.6 with XCode etc. installed. Until now, I was able to get all of the projects to work. 

```
lukasss (stable) homebrew $ brew install ipopt --with-openblas
==> Installing ipopt from brewsci/science
==> Installing dependencies for brewsci/science/ipopt: mumps
==> Tapping dpo/openblas
Cloning into '/usr/local/Homebrew/Library/Taps/dpo/homebrew-openblas'...
remote: Enumerating objects: 34, done.
remote: Counting objects: 100% (34/34), done.
remote: Compressing objects: 100% (32/32), done.
remote: Total 34 (delta 1), reused 13 (delta 0), pack-reused 0
Unpacking objects: 100% (34/34), done.
Tapped 23 formulae (85 files, 125KB).
Error: Formulae found in multiple taps: 
       * dpo/openblas/parmetis
       * brewsci/science/parmetis

Please use the fully-qualified name e.g. dpo/openblas/parmetis to refer the formula.
lukasss (stable) homebrew $ brew untap dpo/openblas    //HERE I'M UNTAPPING DPO OPENBLAS
Untapping dpo/openblas...
Untapped 23 formulae (85 files, 125KB).
lukasss (stable) homebrew $ brew install ipopt --with-openblas
==> Installing ipopt from brewsci/science
==> Installing dependencies for brewsci/science/ipopt: mumps
==> Tapping dpo/openblas
Cloning into '/usr/local/Homebrew/Library/Taps/dpo/homebrew-openblas'...
remote: Enumerating objects: 34, done.
remote: Counting objects: 100% (34/34), done.
remote: Compressing objects: 100% (32/32), done.
remote: Total 34 (delta 1), reused 13 (delta 0), pack-reused 0
Unpacking objects: 100% (34/34), done.
Tapped 23 formulae (85 files, 125KB).
Error: Formulae found in multiple taps: 
       * dpo/openblas/parmetis      // AND HERE THE ERROR IS THROWN AGAIN
       * brewsci/science/parmetis

Please use the fully-qualified name e.g. dpo/openblas/parmetis to refer the formula.
lukasss (stable) homebrew $ 
```

I'm basically stuck... Any hints or ideas?

**Comment by mvirgo** [comment]
Based on the above, I'd suggest trying to remove whichever brew tap you aren't trying to use - the able is for whatever reason including both brewsci and dpo, and brew isn't smart enough to just decide on one or the other.

**Comment by Tsuihao** [comment]
The warning is because you `brew tap` two repositories And both of them has ipopt. 

What you need to do are two steps:
`$ brew untap brewsci/science` // because brewsci/science is not working (at least for me)
`$ brew install ipopt --with-openblas` // now you only have ipopt in dpo, so the warning will be gone!

This should work!

**Comment by thelukasssheee** [comment]
I tried both... as mentioned above, at some later point during installation, brew is always resurrecting the taps which I actually untapped. What could have helped would have been to actually delete the doubled installation routines. 
Strange error.... 

I finally did bypass the issue with a Linux virtual machine. I setup the environment in less than an hour and my code was working instantly.

Thanks for your help though!
